### PR TITLE
[taglib] Update to 2.0

### LINKED ISSUE #36594: [taglib] update to 2.0
### Library name

taglib

### New version number

2.0

### Other information that may be useful (release notes, etc...)

https://github.com/taglib/taglib/releases/tag/v2.0

### PR DESCRIPTION
**Pull request by nlogozzo**
Fixes #36594 

- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [x] SHA512s are updated for each updated download.
- [x] The "supports" clause reflects platforms that may be fixed by this new version.
- [x] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [x] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

Tested on `x64-windows`

### CODE REVIEW

**Comment by LilyWangLL** [comment]
Usage test passed on `x64-windows`.

**Comment by LilyWangLL** [comment]
Please remove the usage file and code `file(INSTALL "${CMAKE_CURRENT_LIST_DIR}/usage" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}")` in portfile.cmake, vcpkg will output the following usage:
```
taglib provides CMake targets:

  # this is heuristically generated, and may not be correct
  find_package(taglib CONFIG REQUIRED)
  target_link_libraries(main PRIVATE TagLib::tag TagLib::tag_c TagLib::TagLib)

taglib provides pkg-config modules:

    # Audio meta-data library
    taglib

    # Audio meta-data library (C bindings)
    taglib_c
```

File: ports/taglib/vcpkg.json [hunk 1]
```diff
@@ -1,15 +1,19 @@
 {
   "name": "taglib",
-  "version-semver": "1.13.1",
-  "port-version": 1,
+  "version-semver": "2.0.0",
```
Comment (LilyWangLL) [inline ports/taglib/vcpkg.json:3]:
```suggestion
  "version": "2.0.0",
```

Comment (LilyWangLL) [reply ports/taglib/vcpkg.json:3]: Is this port a semver library? If it is not, here should use `"version"`?

File: ports/taglib/portfile.cmake [hunk 2]
```diff
@@ -1,8 +1,8 @@
 vcpkg_from_github(
     OUT_SOURCE_PATH SOURCE_PATH
     REPO taglib/taglib
-    REF v1.13.1
-    SHA512 986231ee62caa975afead7e94630d58acaac25a38bc33d4493d51bd635d79336e81bba60586d7355ebc0670e31f28d32da3ecceaf33292e4bc240c64bf00f35b
+    REF v2.0
```
Comment (LilyWangLL) [inline ports/taglib/portfile.cmake:4]:
```suggestion
    REF "v${VERSION}"
```

**Review by LilyWangLL — DISMISSED** [review]

**Comment by nlogozzo** [comment]
fixed all mentioned issues

**Review by data-queue — APPROVED** [review]

**Pull request merged**